packages/fuel-gauge/src/advanced-logging.test.ts

```diff
@@ -100,7 +100,7 @@ describe('Advanced Logging', () => {
   it('can get log data from a downstream Contract', async () => {
     const INPUT = 3;
     const { value, logs } = await contractInstance.functions
-      .test_log_from_other_contract(INPUT, otherContractInstance.id)
+      .test_log_from_other_contract(INPUT, otherContractInstance.id.toB256())
```

Comment:
This is an interesting one, potentially up for discussion.

So the second parameter to this contract function is expecting a `b256` address. Therefore, a `B256Coder` will be instantiated to encode this value. Previously, we would pass in the `contact.id` which is an `Address` type. This was fine previously because `arrayify` which is used inside of the `B256Coder` to encode the value accepts a _hexable_ object as a parameter. This is an object that contains the function `toHexString()`.  So it would be converted to a `b256` by `arrayify`, and then returned as a byte array.

Ethers v6 has much stricter function signatures and does not do such magic.`getBytesCopy` takes `BytesLike` values (essentially a hex string or byte array), therefore we must pass that to the function. Otherwise it cannot be encoded by the `B256Coder` which is now using `getBytesCopy` to return a byte array.

I personally think if we are expecting a `b256` address as defined by the ABI, then that is what should be passed to the function. Rather than doing some augmentation on our side. Unless this is a decision to aid dev ex?

With type inference we would not be able to pass in an `Address` instance when we are expecting a `b256` anyways (without additional magic).

Reply:
I agree with you. Let's make sure we add a small note for this in the release notes though.